numpy/lib/ufunclike.py

```diff
@@ -43,15 +103,11 @@ def fix(x, y=None):
     array([ 2.,  2., -2., -2.])
 
     """
-    x = nx.asanyarray(x)
-    y1 = nx.floor(x)
-    y2 = nx.ceil(x)
-    if y is None:
-        y = nx.asanyarray(y1)
-    y[...] = nx.where(x >= 0, y1, y2)
-    return y
-
-def isposinf(x, y=None):
+    return _where(nx.greater_equal(x, 0), nx.floor(x), nx.ceil(x), out)
```

Comment:
It should be whatever is closest to what a `ufunc` does; any subclasses should be able to deal with `out[()]`... (I guess the alternative would be to separate out the non-array case, but that seems overkill)